src/Libraries/DSCPython/DSCPythonCodeCompletionProviderCore.cs

```diff
@@ -0,0 +1,1277 @@
+        //We do not know if anyone was using that class, but we needed to remove the compile
+        //time references between PythonNodeModels and DSIronPython to support dynamically loading
+        //python versions.
+
+        //note that the public members below are not really public (this is an internal class)
+        //but must be marked that way to satisfy the legacy interface
+        #region BACKING LEGACY CLASS DO NOT MODIFY UNTIL 3
+
+        internal static Dictionary<string, string> FindAllTypeImportStatements(string code)
+        {
+            // matches the following types:
+            //     from lib import *
+
+            var pattern = fromImportRegex +
+                          atLeastOneSpaceRegex +
+                          variableName +
+                          atLeastOneSpaceRegex +
+                          basicImportRegex +
+                          atLeastOneSpaceRegex +
+                          @"\*$";
+
+            var matches = Regex.Matches(code, pattern, RegexOptions.Multiline);
+
+            var importMatches = new Dictionary<string, string>();
+
+            for (var i = 0; i < matches.Count; i++)
+            {
+                var wholeLine = matches[i].Groups[0].Value;
+                var libName = matches[i].Groups[3].Value.Trim();
+
+                if (importMatches.ContainsKey(libName))
+                    continue;
+
+                importMatches.Add(libName, wholeLine);
+            }
+
+            return importMatches;
+        }
+
+        internal static Dictionary<string, string> FindTypeSpecificImportStatements(string code)
+        {
+
+            var pattern = fromImportRegex +
+                          atLeastOneSpaceRegex +
+                          variableName +
+                          atLeastOneSpaceRegex +
+                          basicImportRegex +
+                          atLeastOneSpaceRegex +
+                          commaDelimitedVariableNamesRegex +
+                          "$";
+
+            var matches = Regex.Matches(code, pattern, RegexOptions.Multiline);
+
+            var importMatches = new Dictionary<string, string>();
+
+            for (var i = 0; i < matches.Count; i++)
+            {
+                var wholeLine = matches[i].Groups[0].Value.TrimEnd('\r', '\n');
+                var joinedTypeNames = matches[i].Groups[8].Value.Trim();
+
+                var allTypes = joinedTypeNames.Replace(" ", "").Split(',');
+
+                foreach (var typeName in allTypes)
+                {
+                    if (importMatches.ContainsKey(typeName))
+                    {
+                        continue;
+                    }
+
+                    importMatches.Add(typeName, wholeLine.Replace(joinedTypeNames, typeName));
+                }
+            }
+
+            return importMatches;
+        }
+
+        public static Dictionary<string, string> FindVariableStatementWithRegex(string code, string valueRegex)
```

Comment:
I don't think this needs to be public as it's being obtained by reflection using a non-public flag.

Reply:
Changed it to internal